app/controllers/submissions_controller.rb

```diff
@@ -365,7 +369,17 @@ def update_files
                                 ' for assignment ' +
                                 "'#{@assignment.short_identifier}'.")
         end
-
+        #
+        unless students_filename.length < 1
+          if required_assignments != students_filename &&
+             @assignment.is_required == true
```

Comment:
There should be no need to compare this attribute to "true", since it's always either true or false.